### PR TITLE
Task/routing

### PR DESCRIPTION
**Pull request by AnatoliRub**
Add Routing to the project.

### CODE REVIEW

File: youtube-client/src/app/app.component.html [hunk 1]
```diff
@@ -1,9 +1,2 @@
...
-  (toggleSettings)="toggleSettings()"
-  (showSearchingResults)="showResults($event)"
-></app-header>
-<app-filtering-criteria
-  *ngIf="isShowSettings"
-  (filterData)="getFilterData($event)"
-></app-filtering-criteria>
-<app-login-page></app-login-page>
+<app-header></app-header>
+<router-outlet></router-outlet>
```
Comment (StanevichIvan) [inline youtube-client/src/app/app.component.html:2]: do we have main tag somewhere?

File: youtube-client/src/app/app.module.ts [hunk 2]
```diff
@@ -6,20 +6,12 @@ import { AppRoutingModule } from './app-routing.module';
...
-    BrowserModule,
-    AppRoutingModule,
-    BrowserAnimationsModule,
-    MaterialModule,
-    CoreModule,
-    YoutubeModule,
-    AuthModule,
-  ],
+  declarations: [AppComponent, ErrorPageComponent, ErrorComponent],
+  imports: [AppRoutingModule, BrowserModule, BrowserAnimationsModule, MaterialModule, CoreModule],
```
Comment (StanevichIvan) [inline youtube-client/src/app/app.module.ts:14]: Might animation module could be a part of the CoreModule

File: youtube-client/src/app/auth/pages/registration-page/registration-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  }
+
+  form: FormGroup = new FormGroup({
+    firstName: this.getFirstNameController(),
+    lastName: this.getLastNameController(),
+    email: this.getEmailController(),
+    password: this.getPasswordController(),
+  });
+
+  constructor(private router: Router) {}
```
Comment (StanevichIvan) [inline youtube-client/src/app/auth/pages/registration-page/registration-page.component.ts:51]: please check the functions order. https://palantir.github.io/tslint/rules/member-ordering/

File: youtube-client/src/app/core/components/header/header.component.ts [hunk 4]
```diff
@@ -1,20 +1,14 @@
...
+import { SearchServiceService } from '../../services/search-service.service';
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss'],
 })
 export class HeaderComponent {
-  @Output() toggleSettings = new EventEmitter<void>();
+  public isShow$: Observable<boolean> = this.searchService.isShowSettings;
```
Comment (StanevichIvan) [inline youtube-client/src/app/core/components/header/header.component.ts:11]: please be a bit more specific. specify what is visible

File: youtube-client/src/app/core/services/search-service.service.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { Filter, IFilterData } from 'src/types/filtering-criteria-types';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SearchServiceService {
+  #toggleSearchSettings: boolean = false;
```
Comment (StanevichIvan) [inline youtube-client/src/app/core/services/search-service.service.ts:9]: might TS private works with #

File: youtube-client/src/app/youtube/components/card/card.component.ts [hunk 6]
```diff
@@ -8,10 +10,12 @@ import { Post } from 'src/types/youtube-data';
   styleUrls: ['./card.component.scss'],
 })
 export class CardComponent {
-  @Input() post: Post | undefined;
+  @Input() post?: Post;
```
Comment (StanevichIvan) [inline youtube-client/src/app/youtube/components/card/card.component.ts:13]: Might card should work only with defend post property. Let's set it as required.

File: youtube-client/src/app/app.component.html (hunk 1 above)
Comment (AnatoliRub) [reply youtube-client/src/app/app.component.html]: c7276a3

**Review by StanevichIvan — APPROVED** [review]

File: youtube-client/src/app/app.module.ts (hunk 2 above)
Comment (AnatoliRub) [reply youtube-client/src/app/app.module.ts]: 4d2e554

File: youtube-client/src/app/core/services/search-service.service.ts (hunk 5 above)
Comment (AnatoliRub) [reply youtube-client/src/app/core/services/search-service.service.ts:9]: 77559d7

File: youtube-client/src/app/auth/pages/registration-page/registration-page.component.ts (hunk 3 above)
Comment (AnatoliRub) [reply youtube-client/src/app/auth/pages/registration-page/registration-page.component.ts]: 93e5634

File: youtube-client/src/app/core/components/header/header.component.ts (hunk 4 above)
Comment (AnatoliRub) [reply youtube-client/src/app/core/components/header/header.component.ts]: db5eada

**Pull request merged**